### PR TITLE
feat(crs): support for new CRS v4

### PR DESCRIPTION
**Pull request by fzipi**
Signed-off-by: Felipe Zipitria <felipe.zipitria@owasp.org>

Add new variables in preparation for CRS v4.

### CODE REVIEW

**Comment by fzipi** [comment]
Can someone take a look here and see if I'm not breaking/forgetting anything for the CRS v4 new variables naming?

@theseion @theMiddleBlue @franbuehler

File: docker-compose.yaml [hunk 1]
```diff
@@ -31,6 +31,10 @@ services:
       - ANOMALY_OUTBOUND=4
       # Executing Paranoia Level
       # - EXECUTING_PARANOIA=2
+      #
+      # New variable naming
```
Comment (theseion) [inline docker-compose.yaml:35]:
```suggestion
      # Replaces PARANOIA as of CRS 4
```

File: docker-compose.yaml [hunk 2]
```diff
@@ -31,6 +31,10 @@ services:
...
+      - BLOCKING_PARANOIA_LEVEL=1
+      # - DETECTION_PARANOIA_LEVEL=4
```
Comment (theseion) [inline docker-compose.yaml:37]:
```suggestion
      # Replaces EXECUTING_PARANOIA_LEVEL as of CRS 4
      # - DETECTION_PARANOIA_LEVEL=4
```

File: src/opt/modsecurity/activate-rules.sh [hunk 3]
```diff
@@ -97,7 +107,12 @@ fi
 
 # Substitute MODSEC_TAG
 if [ -n "${MODSEC_TAG}" ]; then
-  sed -z -E -i "s/\\$\{MODSEC_TAG\}/${MODSEC_TAG}/" /etc/modsecurity.d/owasp-crs/crs-setup.conf
+  sed -z -E -i "s/\\$\{MODSEC_TAG\}/${MODSEC_TAG}/g" /etc/modsecurity.d/owasp-crs/crs-setup.conf
+fi
+
+# Reporting Level
+if [ -n "${REPORTING_LEVEL}" ]; then
+  sed -z -E -i 's/#SecAction.{6}id:900115.*tx\.reporting_level=2=1\"/SecAction \\\n  \"id:900115, \\\n   phase:1, \\\n   nolog, \\\n   pass, \\\n   t:none, \\\n   setvar:tx.reporting_level='"${REPORTING_LEVEL}"'\"/' /etc/modsecurity.d/owasp-crs/crs-setup.conf
```
Comment (theseion) [inline src/opt/modsecurity/activate-rules.sh:115]:
```suggestion
  sed -z -E -i 's/#SecAction.{6}id:900115.*tx\.reporting_level=2\"/SecAction \\\n  \"id:900115, \\\n   phase:1, \\\n   nolog, \\\n   pass, \\\n   t:none, \\\n   setvar:tx.reporting_level='"${REPORTING_LEVEL}"'\"/' /etc/modsecurity.d/owasp-crs/crs-setup.conf
```

File: docker-compose.yaml [hunk 4]
```diff
@@ -31,6 +31,10 @@ services:
...
```
Comment (theseion) [inline docker-compose.yaml:38]: Does it make sense to add reporting level here too?

**Review by theseion — CHANGES_REQUESTED** [review]

File: docker-compose.yaml (hunk 1 above)
Comment (fzipi360) [reply docker-compose.yaml]: Added.

File: docker-compose.yaml (hunk 4 above)
Comment (fzipi) [reply docker-compose.yaml:42]: Yes, added.

**Comment by fzipi** [comment]
Thanks @theseion ! Added your suggestions and also described the new variables in the readme.

**Review by theseion — COMMENTED** [review]
LGTM.

What do you think of fully replacing PARANOIA and EXECUTING_PARANOIA? Technically, we don't need the duplication and for users it makes it more complicated, especially if they switch between CRS < 4 and >= 4.
We could document, how the variables relate to the old naming convention, that should be enough.

**Comment by fzipi** [comment]
Merging.

**Comment by fzipi** [comment]
I agree but for now, let's get this rolling and unblock Simon. Pretty sure we will need more changes until we get to a safe state.

We don't want to break other people's flows also. If we remove the variables, 99% sure that many will suffer the change.

**Pull request merged**